domain-models-runtime/src/main/java/org/folio/rest/impl/TenantAPI.java

```diff
@@ -226,40 +264,75 @@ public String sqlFile(String tenantId, boolean tenantExists, TenantAttributes te
     sMaker.setSchemaJson(tableInputStr);
     Schema schema = ObjectMapperTool.getMapper().readValue(tableInputStr, Schema.class);
     sMaker.setSchema(schema);
+    sMaker.setPreviousSchema(previousSchema);
     String sqlFile = sMaker.generateDDL();
     log.debug("GENERATED SCHEMA " + sqlFile);
     return sqlFile;
   }
 
+  private Future<String> sqlFile(Context context, String tenantId, TenantAttributes tenantAttributes,
+      boolean tenantExists) {
+
+    return previousSchema(context, tenantId, tenantExists)
+    .compose(previousSchema -> {
+      try {
+        String sqlFile = sqlFile(tenantId, tenantExists, tenantAttributes, previousSchema);
+        return Future.succeededFuture(sqlFile);
+      } catch (IOException e) {
+        throw new UncheckedIOException(e);
+      } catch (TemplateException e) {  // checked exception from main.tpl parsing
+        throw new IllegalArgumentException(e);
+      }
+    });
+  }
+
+  /**
+   * @param tenantAttributes parameters like module version that may influence generated SQL
+   * @return the SQL commands to create or upgrade the tenant's schema
+   * @throws NoSchemaJsonException when templates/db_scripts/schema.json doesn't exist
+   * @throws TemplateException when processing templates/db_scripts/schema.json fails
+   */
+  public Future<String> sqlFile(Context context, String tenantId, TenantAttributes tenantAttributes) {
+    AtomicBoolean tenantExists = new AtomicBoolean();
+
+    return tenantExists(context, tenantId)
+    .compose(tenantExistsResult -> {
+      tenantExists.set(tenantExistsResult);
+      return previousSchema(context, tenantId, tenantExists.get());
+    }).compose(previousSchema -> {
+      try {
+        String sqlFile = sqlFile(tenantId, tenantExists.get(), tenantAttributes, previousSchema);
+        return Future.succeededFuture(sqlFile);
+      } catch (IOException e) {
+        throw new UncheckedIOException(e);
+      } catch (TemplateException e) {  // checked exception from main.tpl parsing
+        throw new IllegalArgumentException(e);
+      }
+    });
+  }
+
   @Validate
   @Override
-  public void postTenant(TenantAttributes entity, Map<String, String> headers,
+  public void postTenant(TenantAttributes tenantAttributes, Map<String, String> headers,
       Handler<AsyncResult<Response>> handlers, Context context)  {
 
     String tenantId = TenantTool.tenantId(headers);
     log.info("sending... postTenant for " + tenantId);
-    if (entity != null) {
-      log.debug("upgrade from " + entity.getModuleFrom() + " to " + entity.getModuleTo());
+    if (tenantAttributes != null) {
+      log.debug("upgrade from " + tenantAttributes.getModuleFrom() + " to " + tenantAttributes.getModuleTo());
     }
 
```

Comment:
This logs what Okapi passes to it, right? Do we even use moduleFrom or moduleTo for anything mechanical anymore? If so, perhaps we could log what RMB sees as from-to-version?

Reply:
Only recently we started storing the correct version in the rmb_internal table: https://github.com/folio-org/raml-module-builder/commit/3dbf2ed9905bbca558b8b35e25b40115a65fea4a 
Therefore we must use the version number passed in from Okapi for Goldenrod.
SchemaMaker and schema.json use them.

Reply:
ok

Reply:
But this is a possible improvement for Q3?

Reply:
Sure. This is why we store rmb version and schema.json in the db.